src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageProcessor.java

```diff
@@ -0,0 +1,197 @@
+package uk.gov.hmcts.reform.bulkscan.payment.processor.service.servicebus;
+
+import com.google.common.collect.ImmutableMap;
+import com.microsoft.azure.servicebus.IMessage;
+import com.microsoft.azure.servicebus.IMessageReceiver;
+import com.microsoft.azure.servicebus.primitives.ServiceBusException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.bulkscan.payment.processor.client.payhub.PayHubClientException;
+import uk.gov.hmcts.reform.bulkscan.payment.processor.service.servicebus.exceptions.InvalidMessageException;
+import uk.gov.hmcts.reform.bulkscan.payment.processor.service.servicebus.exceptions.MessageProcessingException;
+import uk.gov.hmcts.reform.bulkscan.payment.processor.service.servicebus.handler.MessageProcessingResult;
+import uk.gov.hmcts.reform.bulkscan.payment.processor.service.servicebus.handler.MessageProcessingResultType;
+import uk.gov.hmcts.reform.bulkscan.payment.processor.service.servicebus.handler.PaymentMessageHandler;
+import uk.gov.hmcts.reform.bulkscan.payment.processor.service.servicebus.model.PaymentMessage;
+
+import java.time.Instant;
+
+import static uk.gov.hmcts.reform.bulkscan.payment.processor.service.servicebus.handler.MessageProcessingResultType.POTENTIALLY_RECOVERABLE_FAILURE;
+import static uk.gov.hmcts.reform.bulkscan.payment.processor.service.servicebus.handler.MessageProcessingResultType.SUCCESS;
+import static uk.gov.hmcts.reform.bulkscan.payment.processor.service.servicebus.handler.MessageProcessingResultType.UNRECOVERABLE_FAILURE;
+
+
+@Service
+@Profile("!nosb") // do not register for the nosb (test) profile
+public class PaymentMessageProcessor {
+
+    private static final Logger log = LoggerFactory.getLogger(PaymentMessageProcessor.class);
+
+    private final PaymentMessageHandler paymentMessageHandler;
+    private final IMessageReceiver messageReceiver;
+    private final int maxDeliveryCount;
+
+    public PaymentMessageProcessor(PaymentMessageHandler paymentMessageHandler,
+                                   IMessageReceiver messageReceiver,
+                                   @Value("${azure.servicebus.payments.max-delivery-count}") int maxDeliveryCount) {
+        this.paymentMessageHandler = paymentMessageHandler;
+        this.messageReceiver = messageReceiver;
+        this.maxDeliveryCount = maxDeliveryCount;
+    }
+
+
+    /**
+     * Reads and processes next message from the queue.
+     *
+     * @return false if there was no message to process. Otherwise true.
+     */
+    public boolean processNextMessage() throws ServiceBusException, InterruptedException {
+        IMessage message = messageReceiver.receive();
```

Comment:
This active approach to reading messages from the queue might seem a bit controversial, but I think it's good - makes it possible to introduce circuit breaker, e.g. for times when PayHub is down or sth.